packages/transform-module/README.md

```diff
@@ -1,5 +1,200 @@
+quuux = 'Hello, World!';
+```
+
+From this, the analyzer produces a module static record that shows what modules
+the module needs to be linked to and how to link their imports and exports.
+
+```js
+{
+  "exportAlls": [ "import-and-export-all.js" ],
+  "imports": {
+    "import-default-export-from-me.js": [ "default" ],
+    "import-all-from-me.js": [ "*" ],
+    "import-named-exports-from-me.js": [ "fizz", "buzz" ],
+    "import-named-export-and-rename.js": [ "color" ],
+    "import-and-reexport-name-from-me.js": [ "qux" ],
+    "import-and-export-all.js": []
+  },
+  "liveExportMap": {
+    "qux": [ "qux", false ],
+    "quuux": [ "quuux", true ],
+  },
+  "fixedExportMap": {
+    "default": [ "default" ],
+    "quux": [ "quux" ],
+  }
+}
+```
+
+The functor source, the module transformed into a program, has the following
+shape.
+The names are additionally obscured with Unicode zero-width-joiners to avoid
+collisions with sensibly constructed modules, and the transformation preserves
+line numbers.
+
+```js
+(({
+  imports: $h_imports,
+  liveVar: $h_live,
+  onceVar: $h_once,
+}) => {
+  let foo, bar, fizz, buzz, colour;
+  $h_imports(
+    new Map([
+      ["import-default-export-from-me.js", new Map([
+        ["default", [$h_a => (foo = $h_a)]],
+      ])],
+      ["import-all-from-me.js", new Map([
+        ["*", [$h_a => (bar = $h_a)]]
+      ])],
+      ["import-named-exports-from-me.js", new Map([
+        ["fizz", [$h_a => (fizz = $h_a)]],
+        ["buzz", [$h_a => (buzz = $h_a)]],
+      ])],
+      ["import-named-export-and-rename.js", new Map([
+        ["color", [$h_a => (colour = $h_a)]],
+      ])],
+      ["import-and-reexport-name-from-me.js", new Map([
+        ["qux", [$h_live["qux"]]]
+      ])],
+      ["import-and-export-all.js", new Map([])]
+    ]),
+    ["import-and-export-all.js"]
+  );
+
+  let $c_quuux = null;
+  $h_live.quuux($c_quuux);
+
+  const { default: $c_default } = { default: 42 };
+
+  $h_once.default($c_default);
+
+  const quux = 'Hello, World!';
+  $h_once.quux(quux);
+
+  quuux = 'Sorry for binding late!';
+})
+```
+
+For the final live binding to `quuux`, we depend on the evaluator to put a
+Proxy on the scope chain to intercept the assignment and effect an update
+to all modules that import the value.
+
+```ts
+// This is the signature of the analyze function, after composing it
+// with Babel core.
+type Analyzer = ({string: ModuleSource}) => ModuleStaticRecord
+
+type ModuleSource = string
+
+type ModuleStaticRecord = {
+  exportAlls: ExportAlls,
+  imports: Imports,
+  liveExportMap: LiveExportMap,
+  fixedExportMap: FixedExportMap,
+  functorSource: string,
+};
+
+// ExportAlls are the relative module specifiers found in directives like:
+//   export * from 'import-and-reexport-all-from-me.js';
+// These are both on the module static record and passed to the import function
+// injected into a module functor.
+// TODO remove the import argument. It does not appear to be used
```

Comment:
What does this TODO mean?

Reply:
I found that the `import` function injected into the functor by `makeModuleInstance` does not actually accept the `exportAlls` argument. It uses `exportAlls` on the linkage record instead. So, my guess is transform-module doesn’t need to thread that argument.